public/app/features/dashboard/state/DashboardMigrator.ts

```diff
@@ -621,18 +621,27 @@ export class DashboardMigrator {
     }
 
     if (oldVersion < 27) {
-      for (const variable of this.dashboard.templating.list) {
+      this.dashboard.templating.list = this.dashboard.templating.list.map((variable) => {
         if (!isConstant(variable)) {
-          continue;
+          return variable;
         }
 
-        if (variable.hide === VariableHide.dontHide || variable.hide === VariableHide.hideLabel) {
-          variable.type = 'textbox';
+        const newVariable: ConstantVariableModel | TextBoxVariableModel = {
+          ...variable,
+        };
+
+        newVariable.current = { selected: true, text: newVariable.query ?? '', value: newVariable.query ?? '' };
+        newVariable.options = [newVariable.current];
+
+        if (newVariable.hide === VariableHide.dontHide || newVariable.hide === VariableHide.hideLabel) {
+          return {
+            ...newVariable,
+            type: 'textbox',
+          };
```

Comment:
I am incredibly uneasy about this!!!! 

The problem is that you cannot _change_ the type of an interface with typescript, but we cannot change the identity of the `variable` variable in a for-loop, because we need to mutate it.